File: Erik-BabylonJS-Test/Assets/Scripts/BabylonTest/BTestComponent.ts
module PROJECT {
	/**
	 * Babylon Script Component
	 * @class BTestComponent
	 */
	export class BTestComponent extends BABYLON.ScriptComponent {
		// Example: private helloWorld:string = "Hello World";

		private val1: number = -1;
		private val2: number = -1;
		private val3: boolean = false;

		protected awake(): void {
			/* Init component function */

			this.val1 = this.getProperty('testVal_1');
			this.val2 = this.getProperty('testVal_2');
			this.val3 = this.getProperty('testVal_3');
		}

		protected start(): void {
			/* Start render loop function */
		}

		protected ready(): void {
			/* Execute when ready function */
		}

		protected update(): void {
			/* Update render loop function */
		}

		protected late(): void {
			/* Late update render loop function */
		}

		protected after(): void {
			/* After update render loop function */
		}

		protected fixed(): void {
			/* Fixed update physics step function */
		}

		protected destroy(): void {
			/* Destroy component function */
		}

		public printValues() {
			console.log(`Test Component Values - 1: ${this.val1}  2: ${this.val2}  3: ${this.val3}`);
		}
	}
}

File: Erik-BabylonJS-Test/Assets/Scripts/BabylonTest/BTestComponent.cs
using System;
using UnityEditor;
using UnityEngine;

/**
* Editor Script Component
* @class BTestComponent
*/
[Babylon(Class = "PROJECT.BTestComponent"), AddComponentMenu("Scripts/My Project/BTestComponent")]
public class BTestComponent : EditorScriptComponent
{
    /* Add Editor Properties To Script Component */
    // Example: [Tooltip("Example hello world property")]
    // Example: [Auto] public string helloWorld = "Hello World";

    public float testVal_1 = 1.0f;
    public float testVal_2 = 2.0f;
    public bool testVal_3 = true;

    /* [Serializable, HideInInspector] public string exportProperty = null; */
    public override void OnUpdateProperties(Transform transform, SceneExporterTool exporter)
    {
        // Example: this.helloWorld = "Update Hello World";
    }
}

// Optional Script Component Custom Editor Class
[CustomEditor(typeof(BTestComponent)), CanEditMultipleObjects]
public class BTestComponentEditor : Editor
{
    public void OnEnable()
    {
        BTestComponent owner = (BTestComponent)target;
    }
    public override void OnInspectorGUI()
    {
        base.OnInspectorGUI();
    }
}